usaspending_api/etl/management/commands/load_submission.py

```diff
@@ -660,14 +660,20 @@ def load_file_d1(submission_attributes, procurement_data, db_cursor):
         pop_location, created = get_or_create_location(
             place_of_performance_field_map, row, place_of_performance_value_map)
 
-        # If awarding/funding toptier agency code (aka CGAC) is not supplied on the D1 record,
-        # use the sub tier code to look it up
-        if row['awarding_agency_code'] is None:
+        # If awarding toptier agency code (aka CGAC) is not supplied on the D1 record,
+        # use the sub tier code to look it up. This code assumes that all incoming
+        # records will supply an awarding subtier agency code
+        if row['awarding_agency_code'] is None or len(row['awarding_agency_code'].strip()) < 1:
```

Comment:
Hmm, for a future improvement, it would be nice to strip all fields at once upon load from the file instead of having to remember to do so at various points during processing...